tooling/windows_repro_process.sh

```diff
@@ -42,69 +32,30 @@ if [[ ! -d "${JDK_DIR}" ]] || [[ ! -d "${JDK_DIR}/bin"  ]]; then
   exit 1
 fi
 
-echo "Expanding the 'modules' Image to remove signatures from within.."
-jimage extract --dir "${JDK_DIR}/lib/modules_extracted" "${JDK_DIR}/lib/modules"
-rm ${JDK_DIR}/lib/modules
-
-echo "Expanding the 'src.zip' to normalize file permissions"
-unzip ${JDK_DIR}/lib/src.zip -d ${JDK_DIR}/lib/src_zip_expanded
-rm ${JDK_DIR}/lib/src.zip
-
-echo "Expanding jmods to process exe/dll within"
-FILES=$(find "${JDK_DIR}" -type f -path '*.jmod')
-for f in $FILES
-  do
-    echo "Unzipping $f"
-    base=$(basename $f)
-    dir=$(dirname $f)
-    expand_dir="${dir}/expanded_${base}"
-    mkdir -p "${expand_dir}"
-    jmod extract --dir "${expand_dir}" "$f"
-    rm "$f"
-  done
-
-echo "Expanding the 'jrt-fs.jar' to remove signatures from within.."
-mkdir ${JDK_DIR}/lib/jrt-fs-expanded
-unzip -d ${JDK_DIR}/lib/jrt-fs-expanded ${JDK_DIR}/lib/jrt-fs.jar
-rm ${JDK_DIR}/lib/jrt-fs.jar
-
-mkdir ${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs-expanded
-unzip -d ${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs-expanded ${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs.jar
-rm ${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs.jar
+expandJDK "$JDK_DIR"
 
 # Remove existing signature
-removeSignatures
+removeSignatures "$JDK_DIR" "$OS"
 
-# Add the SELF_SIGN temporary signature
-echo "Adding SELF_SIGN Signatures for ${JDK_DIR}"
-FILES=$(find "${JDK_DIR}" -type f -path '*.exe' && find "${JDK_DIR}" -type f -path '*.dll')
-for f in $FILES
-  do
-    echo "Signing $f"
-    if signtool sign /f $SELF_CERT_FILE /p $SELF_CERT_PASS $f ; then
-        echo "  ==> Successfully signed $f"
-    else
-        echo "  ==> $f failed to be signed!!"
-        exit 1
-    fi
-  done
-
-echo "Successfully SELF_CERT signed all Signatures in ${JDK_DIR}"
+# Sign with temporary Signature, which when removed results in determinisitic binary length
+tempSign "$JDK_DIR" "$OS" "$SELF_CERT_FILE" "$SELF_CERT_PASS"
 
 # Remove temporary SELF_SIGN signature, which will then normalize binary length
-removeSignatures
-
-echo "Removing jrt-fs.jar MANIFEST.MF BootJDK vendor string lines"
-sed -i '/^Implementation-Vendor:.*$/d' "${JDK_DIR}/lib/jrt-fs-expanded/META-INF/MANIFEST.MF"
-sed -i '/^Created-By:.*$/d' "${JDK_DIR}/lib/jrt-fs-expanded/META-INF/MANIFEST.MF"
-sed -i '/^Implementation-Vendor:.*$/d' "${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs-expanded/META-INF/MANIFEST.MF"
-sed -i '/^Created-By:.*$/d' "${JDK_DIR}/jmods/expanded_java.base.jmod/lib/jrt-fs-expanded/META-INF/MANIFEST.MF"
-
-echo "Removing SOURCE= from ${JDK_DIR}/release file, as Temurin builds from mirror _adopt tag"
-sed -i '/^SOURCE=.*$/d' "${JDK_DIR}/release"
-
-echo "Removing cacerts file, as Temurin builds with different Mozilla cacerts"
-find "${JDK_DIR}" -type f -name "cacerts" -delete
+removeSignatures "$JDK_DIR" "$OS"
+
+if [[ $(uname) =~ Darwin* ]]; then
```

Comment:
So the intention in general is to be able to run these scripts anywhere, and the sed commands need a different syntax on Mac, but the reality in the above is the removeSignatues "$JDK_DIR" "CYGWIN" , above it can only run on Windows as it has to use "signtool" from MSVS !
I was a bit over eager with the copy&paste!

Reply:
moved into new function, as per suggestion below...